de.metas.business/src/main/java/de/metas/payment/api/impl/AbstractPaymentDAO.java

```diff
@@ -181,4 +188,68 @@ public BigDecimal getInvoiceOpenAmount(I_C_Payment payment, final boolean credit
 				.listIds(PaymentId::ofRepoId)
 				.stream();
 	}
+	@Override
+	public void updateDiscountAndPayment(I_C_Payment payment,int c_Invoice_ID, I_C_DocType c_DocType)
```

Comment:
please add an abstract method `updateDiscountAndPayment` 6to this class, and 
* move the implementation to `PaymentDAO`
* also add an implementation to `PlainPaymentDAO` (which dan for now just throw an apache-commons `NotImplementedException`)

Background: when running in unit test mode, Services.get() will by default return `PlainPaymentDAO`, which can contain some mockup code that doesn't connect to DB.